File: news/src/main/java/com/leory/badminton/news/mvp/ui/activity/HandOffRecordActivity.java
package com.leory.badminton.news.mvp.ui.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.leory.badminton.news.R;
import com.leory.badminton.news.R2;
import com.leory.badminton.news.di.component.DaggerHandOffRecordComponent;
import com.leory.badminton.news.mvp.contract.HandOffRecordContract;
import com.leory.badminton.news.mvp.model.bean.HandOffBean;
import com.leory.badminton.news.mvp.presenter.HandOffRecordPresenter;
import com.leory.badminton.news.mvp.ui.adapter.HandOffRecordAdapter;
import com.leory.commonlib.base.BaseActivity;
import com.leory.commonlib.base.delegate.IComponent;
import com.leory.commonlib.di.component.AppComponent;
import com.leory.commonlib.di.scope.ActivityScope;
import com.leory.commonlib.image.ImageLoader;
import com.leory.commonlib.utils.ImageUtils;
import com.leory.commonlib.utils.ToastUtils;
import com.leory.commonlib.widget.XSDToolbar;

import java.util.ArrayList;

import butterknife.BindView;

/**
 * Describe : 交手记录
 * Author : leory
 * Date : 2019-07-23
 */
@ActivityScope
public class HandOffRecordActivity extends BaseActivity<HandOffRecordPresenter> implements HandOffRecordContract.View {

    private static final String KEY_HAND_OFF_URL = "key_hand_off_url";
    @BindView(R2.id.toolbar)
    XSDToolbar toolbar;
    @BindView(R2.id.rcv)
    RecyclerView rcv;
    View player11,player12,player21,player22;
    ImageView head11,head12,head21,head22;
    ImageView flag11,flag12,flag21,flag22;
    TextView name11,name12,name21,name22;
    TextView ranking1,ranking2;
    TextView score1,score2;

    private HandOffRecordAdapter adapter;
    public static void launch(Activity preActivity, String handOffUrl) {
        preActivity.startActivity(new Intent(preActivity, HandOffRecordActivity.class).putExtra(KEY_HAND_OFF_URL, handOffUrl));
    }

    @Override
    public IComponent setupActivityComponent(IComponent component) {
        DaggerHandOffRecordComponent.builder()
                .view(this)
                .recordUrl(getIntent().getStringExtra(KEY_HAND_OFF_URL))
                .appComponent((AppComponent) component)
                .build()
                .inject(this);
        return null;
    }

    @Override
    public int initView(@Nullable Bundle savedInstanceState) {
        return R.layout.activity_hand_off;
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        toolbar.setOnBackListener(() -> finish());
        rcv.setLayoutManager(new LinearLayoutManager(this));
        rcv.setAdapter(adapter=new HandOffRecordAdapter(new ArrayList<>()));
        View head=LayoutInflater.from(this).inflate(R.layout.head_hand_off,null);
        player11=head.findViewById(R.id.player11);
        player12=head.findViewById(R.id.player12);
        player21=head.findViewById(R.id.player21);
        player22=head.findViewById(R.id.player22);
        head11=head.findViewById(R.id.head11);
        head12=head.findViewById(R.id.head12);
        head21=head.findViewById(R.id.head21);
        head22=head.findViewById(R.id.head22);
        flag11=head.findViewById(R.id.flag11);
        flag12=head.findViewById(R.id.flag12);
        flag21=head.findViewById(R.id.flag21);
        flag22=head.findViewById(R.id.flag22);
        name11=head.findViewById(R.id.player11_name);
        name12=head.findViewById(R.id.player12_name);
        name21=head.findViewById(R.id.player21_name);
        name22=head.findViewById(R.id.player22_name);
        ranking1=head.findViewById(R.id.player1_rank);
        ranking2=head.findViewById(R.id.player2_rank);
        score1=head.findViewById(R.id.score1);
        score2=head.findViewById(R.id.score2);
        adapter.addHeaderView(head);
    }

    @Override
    public void showMessage(@NonNull String message) {
        ToastUtils.showShort(message);
    }

    @Override
    public void showHandOffView(HandOffBean bean) {
        if(TextUtils.isEmpty(bean.getPlayer12Name())){
            player12.setVisibility(View.GONE);
        }
        if(TextUtils.isEmpty(bean.getPlayer22Name())){
            player22.setVisibility(View.GONE);
        }
        name11.setText(bean.getPlayer1Name());
        name12.setText(bean.getPlayer12Name());
        name21.setText(bean.getPlayer2Name());
        name22.setText(bean.getPlayer22Name());
        score1.setText(bean.getPlayer1Win());
        score2.setText(bean.getPlayer2Win());
        ranking1.setText(bean.getPlayer1Ranking());
        ranking2.setText(bean.getPlayer2Ranking());
        ImageUtils.loadImage(this,head11,bean.getPlayer1HeadUrl());
        ImageUtils.loadImage(this,head12,bean.getPlayer12HeadUrl());
        ImageUtils.loadImage(this,head21,bean.getPlayer2HeadUrl());
        ImageUtils.loadImage(this,head22,bean.getPlayer22HeadUrl());
        ImageUtils.loadImage(this,flag11,bean.getPlayer1Flag());
        ImageUtils.loadImage(this,flag12,bean.getPlayer12Flag());
        ImageUtils.loadImage(this,flag21,bean.getPlayer2Flag());
        ImageUtils.loadImage(this,flag22,bean.getPlayer22Flag());
        adapter.setNewData(bean.getRecordList());

    }
}

File: news/src/main/java/com/leory/badminton/news/mvp/ui/activity/HandOffRecordActivity.kt
package com.leory.badminton.news.mvp.ui.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import com.leory.badminton.news.R
import com.leory.badminton.news.di.component.DaggerHandOffRecordComponent
import com.leory.badminton.news.mvp.contract.HandOffRecordContract
import com.leory.badminton.news.mvp.model.bean.HandOffBean
import com.leory.badminton.news.mvp.presenter.HandOffRecordPresenter
import com.leory.badminton.news.mvp.ui.adapter.HandOffRecordAdapter
import com.leory.commonlib.base.BaseActivity
import com.leory.commonlib.base.delegate.IComponent
import com.leory.commonlib.di.component.AppComponent
import com.leory.commonlib.di.scope.ActivityScope
import com.leory.commonlib.utils.ImageUtils
import com.leory.commonlib.utils.ToastUtils
import kotlinx.android.synthetic.main.activity_hand_off.*
import kotlinx.android.synthetic.main.head_hand_off.*
import java.util.*

/**
 * Describe : 交手记录
 * Author : leory
 * Date : 2019-07-23
 */
@ActivityScope
class HandOffRecordActivity : BaseActivity<HandOffRecordPresenter>(), HandOffRecordContract.View {

    companion object {

        private const val KEY_HAND_OFF_URL = "key_hand_off_url"

        @JvmStatic
        fun launch(preActivity: Activity, handOffUrl: String?) {
            preActivity.startActivity(Intent(preActivity, HandOffRecordActivity::class.java).putExtra(KEY_HAND_OFF_URL, handOffUrl))
        }
    }

    val adapter: HandOffRecordAdapter by lazy { HandOffRecordAdapter(ArrayList()) }

    override fun setupActivityComponent(component: IComponent): IComponent? {
        DaggerHandOffRecordComponent.builder()
                .view(this)
                .recordUrl(intent.getStringExtra(KEY_HAND_OFF_URL))
                .appComponent(component as AppComponent)
                .build()
                .inject(this)
        return null
    }

    override fun initView(savedInstanceState: Bundle?): Int {
        return R.layout.activity_hand_off
    }

    override fun initData(savedInstanceState: Bundle?) {
        toolbar.setOnBackListener { finish() }
        rcv.layoutManager = LinearLayoutManager(this)
        rcv.adapter = adapter
        val head = LayoutInflater.from(this).inflate(R.layout.head_hand_off, null)
        adapter.addHeaderView(head)
    }

    override fun showMessage(message: String) {
        ToastUtils.showShort(message)
    }

    override fun showLoading() {
        loading.visibility = View.VISIBLE
    }

    override fun hideLoading() {
        loading.visibility = View.GONE
    }

    override fun showHandOffView(bean: HandOffBean) {
        if (TextUtils.isEmpty(bean.player12Name)) {
            player12.visibility = View.GONE
        }
        if (TextUtils.isEmpty(bean.player22Name)) {
            player22.visibility = View.GONE
        }
        player11_name.text = bean.player1Name
        player12_name.text = bean.player12Name
        player21_name.text = bean.player2Name
        player22_name.text = bean.player22Name
        score1.text = bean.player1Win
        score2.text = bean.player2Win
        player1_rank.text = bean.player1Ranking
        player2_rank.text = bean.player2Ranking
        ImageUtils.loadImage(this, head11, bean.player1HeadUrl)
        ImageUtils.loadImage(this, head12, bean.player12HeadUrl)
        ImageUtils.loadImage(this, head21, bean.player2HeadUrl)
        ImageUtils.loadImage(this, head22, bean.player22HeadUrl)
        ImageUtils.loadImage(this, flag11, bean.player1Flag)
        ImageUtils.loadImage(this, flag12, bean.player12Flag)
        ImageUtils.loadImage(this, flag21, bean.player2Flag)
        ImageUtils.loadImage(this, flag22, bean.player22Flag)
        adapter.setNewData(bean.recordList)

    }


}
